server/eclipse-project/src/main/java/us/freeandfair/corla/endpoint/CVRExportImport.java

```diff
@@ -171,14 +171,14 @@ private void parseFile(final Response the_response, final UploadedFile the_file)
         Main.LOGGER.info(imported + " CVRs parsed from file " + the_file.id() + 
                          " for county " + the_file.county().id());
         updateCountyDashboard(the_response, the_file, ImportStatus.SUCCESSFUL, imported);
-        the_file.setStatus(FileStatus.IMPORTED_AS_CVR_EXPORT);
-        Persistence.saveOrUpdate(the_file);
         handleTies(the_response, the_file.county());
         final Map<String, Integer> response = new HashMap<String, Integer>();
         response.put("records_imported", imported);
         if (deleted > 0) {
           response.put("records_deleted", deleted);
         }
+        the_file.setStatus(FileStatus.IMPORTED_AS_CVR_EXPORT);
```

Comment:
Why does moving this here make a difference wrt transactions? Were we e.g. sometimes running into database contention in `handleTies()`, and then when the calling code retried, the file update was spuriously appearing to be done already?

Reply:
The short answer is yes, we had contention in `handleTies` (one of the only places where that happens), and so the file status would revert and not be reset. Now it gets committed with the overall transaction. Though that changes again with the next PR.